### DISCUSSION TITLE
Testing Opportunity

### DISCUSSION THREAD

**Discussion by paulobreim** [sepinf-inc/IPED#1886] [General]
I have a new case to work on that involves indexing approximately 45 ufdr images using IPED.
My idea is to index them all at once.
I'm going to put them all in a single folder and just use -d \folder so I don't have to specify them one by one.
Since all images are cell phones, my idea is to use voice transcription
This will be a very interesting test.
If anyone wants to take the opportunity to test other items available on IPED, just tell me.

**Comment by wladimirleite** [comment]
Hi @paulobreim!
Just one important detail, processing a folder with UFDRs isn't the correct way.
You need to use -d for each UFDR. What I usually do is to write a trivial script/program to list files and generate the command line.
This has been discussed a few times, but I can't remember where :-)
For the future, this may be supported, perhaps through a new command line option to process all evidences (E01, DD, UFDRs) inside a folder or a mask (like -d f:\Case001\*.ufdr).

**Reply by paulobreim** [reply]
@tc-wleite humm Thank you for the information. 
I will use a lot of -d and if it is not supported on the command line, I can split it into more than one processing using append.

**Reply by wladimirleite** [reply]
Not sure if I was clear...
Multiple UFDRs can be processed in a single command (considering that we have 2 UFDRs - 1.ufdr, 2.ufdr and 3.ufdr - inside the folder D:\XX):
`iped.exe -d D:\XX\1.ufdr -d D:\XX\2.ufdr -d D:\XX\3.ufdr -o D:\Case1`
What does NOT work is:
`iped.exe -d D:\XX -o D:\Case1`
In fact it "works" (no error), but the UFDR content won't be handled as it should, just as a general compressed file.
While processing UFDRs inside a folder is not supported, I suggested to abort with an error, as this is a common source of confusion.

**Reply by paulobreim** [reply]
You were clear, but I didn't know what character limit is supported by cmd. A few minutes ago I checked this limit on the Microsoft website and they reported that the limit is 8,191 characters. That's enough.
tks

**Reply by lfcnassif** [reply]
Months ago a notice about UFDR processing was published here: https://github.com/sepinf-inc/IPED/discussions/1272

**Reply by lfcnassif** [reply]
One of the reasons for the behavior change was this situation of several UFDR files inside the input folder, just one of them would be processed fine by IPED 3. You should pass each of them to `-d` param to process them correctly in IPED 4, as @tc-wleite said.

**Reply by paulobreim** [reply]
I started indexing 42 UFDRs but after several exceptions in different UFDRs (I removed the ones that gave errors), I decided to carry out tests with one of those that gave errors.
The error is this:
java.lang.Exception: Error decoding datasource j:\teste.ufdr
at iped.engine.datasource.ItemProducer.run(ItemProducer.java:154)
Caused by: java.lang.RuntimeException: Invalid UFDR file j:\teste.ufdr
at iped.engine.datasource.UfedXmlReader.getXmlInputStream(UfedXmlReader.java:178)
at iped.engine.datasource.UfedXmlReader.lookUpXmlReportInputStream(UfedXmlReader.java:193)
at iped.engine.datasource.UfedXmlReader.read(UfedXmlReader.java:220)
at iped.engine.datasource.UfedXmlReader.read(UfedXmlReader.java:209)
at iped.engine.datasource.ItemProducer.run(ItemProducer.java:123)

I started checking all the ones that gave errors and found one thing common in all of them.
These UFDRs came from a real case and I don't know how these extractions were carried out with Cellebrite.
There are normally 2 files, as in the example below:
![image](https://github.com/sepinf-inc/IPED/assets/381594/a709e8e5-ede4-4b37-9c39-5e1d24b9d257)

But the ones that give errors have more files, some with several files much larger than the ufdr itself.
see these 2 examples:

![image](https://github.com/sepinf-inc/IPED/assets/381594/71fc1841-c879-4ee7-be63-a9cab02a6386)
![image](https://github.com/sepinf-inc/IPED/assets/381594/94e59f7b-6a50-4383-970e-08282b448582)

Looking at the files they look the same, although the hash is not.
![image](https://github.com/sepinf-inc/IPED/assets/381594/73d87c64-1921-4193-8d9c-5d7a361a2ab1)

The ufdr files don't open with 7z because they really aren't a zip, they're something very strange.
Ve
![image](https://github.com/sepinf-inc/IPED/assets/381594/8426b42c-25c0-4b92-8d6f-72311ff46129)

But if cellebrite opens it, it shouldn't be the .ufdr that it is reading.

Has anyone ever seen a situation like this?

After I removed these files from processing, the other 35 that remained seem to be doing well, so far.
But obviously it will take a long time.
![image](https://github.com/sepinf-inc/IPED/assets/381594/193cc477-542a-41d4-8bf5-df5681bb9058)

**Reply by lfcnassif** [reply]
Hi @paulobreim. Those UFDR files are actually segmented ZIP files with a special content. If IPED complains they are invalid, maybe some segment is corrupted (or missing). I suggest you renaming the main file (the UFDR) to ZIP and to try to uncompress it using 7zip, it should extract the content of all Z01, Z02, etc files automatically and warn if some is corrupted.

**Reply by paulobreim** [reply]
Hi @lfcnassif ,
I tried, but the file ufdr show that is not a zip file.

See that the first lines are a txt and are still in a strange format. The first line has this </value> which indicates that there should be things like a <value> beforehand.
After a certain bit this txt has binary stuff.

In this case there are 3 zips very big.
![image](https://github.com/sepinf-inc/IPED/assets/381594/95c19bcd-d6bf-417d-89fe-534197bfe0a2)


What makes me curious is how cellebrite is managing to treat these files. And in this example it opens the 3 zips to work.
I need more time to be able to analyze this with more study.

Thank you for you attention.

**Comment by paulobreim** [comment]
I did some more tests and cellebrite really has something strange.
Although the .ufdr file appears to be invalid, even starting with a piece of txt files, when I run cellebrite it opens the file .z01 file and runs correctly.
